### PR TITLE
For #8312 - Show Top Sites in a ViewPager

### LINKED ISSUE #8312: [Bug] There is no limit of how many top-sites you can add
**Update from UX:**

- To keep the design scalable, let's limit top sites to 16 (2 x 8 swipe-able, see screenshot and please ignore the pin icon on the first three tabs for now)
- Once people add their 9th top site we show dots below top sites to indicate that there's  more 
- People can swipe (similar to how it works in Fennec) to see their 9th-16th top site
- When people are about to add a 17th top site a dialog is displayed telling them that they reached the max amount for top sites 

image

-------
## Steps to reproduce
1. Launch Fenix.
2. Add 50+ top-sites.

### Expected behavior
There should be a limit regarding how many top-sites you can add.
### Actual behavior
50+ top-sites can be added.
### Device information

* Android device: **Samsung Galaxy S10+(Android 9)**
* Fenix version: **Latest Nightly from 2/11.**

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #8312 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
This follows the approach that is used in Firefox Lite https://github.com/mozilla-tw/FirefoxLite/commit/7d7be8ec16d7b6894993db7f466fc3069b7f0300. The Top site layout is as follow:
`HomeFragment` RecyclerView -> ViewPager2 (displays the individual pages)  -> RecyclerView (4 column grid) -> TopSiteItems

The `PagerIndicator` is reused from https://github.com/mozilla-tw/FirefoxLite/blob/32ddedf67826e13ae0c6dd3591117f5020daebd4/app/src/main/java/org/mozilla/rocket/home/topsites/ui/PagerIndicator.kt. 

We removed the dependency on `google_flexbox` since that is no longer used. 

The proper layout of the items and grid will be done in https://github.com/mozilla-mobile/fenix/issues/13765.

File: app/src/main/res/color/pager_dot.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:color="@color/light_grey_50" android:state_selected="true" />
+    <item android:color="@color/photonLightGrey30" />
```
Comment (ekager) [inline app/src/main/res/color/pager_dot.xml:7]: do we want these to be attrs for light/dark ?

File: app/src/main/res/drawable/pager_dot.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="oval">
+    <solid android:color="@color/pager_dot"/>
```
Comment (ekager) [inline app/src/main/res/drawable/pager_dot.xml:7]: same q here about attrs

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TopSitePagerViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        // Don't show any page indicator if there is only 1 page.
+        val numPages = if (topSites.size > TOP_SITES_PER_PAGE) {
+            TOP_SITES_MAX_PAGE_SIZE
+        } else {
+            0
+        }
+
+        if (numPages > 1) {
+            pageIndicator.visibility = View.VISIBLE
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TopSitePagerViewHolder.kt:48]: Nice simplification!

File: app/src/main/res/color/pager_dot.xml (hunk 1 above)
Comment (gabrielluong) [reply app/src/main/res/color/pager_dot.xml:7]: This is a great catch, thank you!

Comment (gabrielluong) [reply app/src/main/res/color/pager_dot.xml]: Fixed.

Comment (ekager) [reply app/src/main/res/color/pager_dot.xml]: Sorry I didn't mean just name the color . Did we want a different color for dark vs light? Like `?primaryText` Or did the UX mocks have `photonLightGrey30` for both light and dark.

Comment (gabrielluong) [reply app/src/main/res/color/pager_dot.xml]: Should be fixed in the latest. Colors came over Slack - commented here https://github.com/mozilla-mobile/fenix/issues/8312#issuecomment-679292000.

Comment (ekager) [reply app/src/main/res/color/pager_dot.xml]: oh i can't read nvm lol this is ready!

**Pull request merged**